### PR TITLE
Add shadow, outline, glow filter support for UITextField

### PR DESCRIPTION
**Pull request by zhouxiaoxiaoxujian**

### CODE REVIEW

File: cocos/ui/UIText.cpp [hunk 1]
```diff
@@ -336,6 +336,24 @@ void Text::updateTextureRGBA()
 {
     updateRGBAToRenderer(_labelRenderer);
 }
+    
+void Text::enableShadow(const Color4B& shadowColor,const Size &offset, int blurRadius) {
+    _labelRenderer->enableShadow(shadowColor, offset, blurRadius);
+}
+
+void Text::enableOutline(const Color4B& outlineColor,int outlineSize) {
+    if (_type == Type::TTF)
```
Comment (zilongshanren) [inline cocos/ui/UIText.cpp]: I think you should remove the if statement here. The Label class does this checking when it is a system font.
And on iOS and Android platform, system font does support Outline. please remove this checking.

File: cocos/ui/UIText.h [hunk 2]
```diff
@@ -158,6 +158,22 @@ class Text : public Widget
...
     TextVAlignment getTextVerticalAlignment();
     
+    /**
+     * Enable shadow for the label
+     *
+     * @todo support blur for shadow effect
+     */
+    void enableShadow(const Color4B& shadowColor = Color4B::BLACK,const Size &offset = Size(2,-2), int blurRadius = 0);
+    
+    /** only support for TTF */
```
Comment (zilongshanren) [inline cocos/ui/UIText.h]: This comment is misleading, The system font also support outline, but only works on iOS and Android

**Comment by zilongshanren** [comment]
@zhouxiaoxiaoxujian 
please modify the PR according to my comments, thanks.

**Comment by zhouxiaoxiaoxujian** [comment]
I have modified, please check PR, thanks.

**Comment by zilongshanren** [comment]
@minggo 
This PR could be merged now.

**Comment by zilongshanren** [comment]
this PR is merged via https://github.com/cocos2d/cocos2d-x/pull/7014

**Pull request closed without merging**